src/en/gogoanime/src/eu/kanade/tachiyomi/animeextension/en/gogoanime/GogoAnime.kt

```diff
@@ -187,17 +160,15 @@ class GogoAnime : ConfigurableAnimeSource, ParsedAnimeHttpSource() {
         val document = response.asJsoup()
         val hosterSelection = preferences.getStringSet(PREF_HOSTER_KEY, PREF_HOSTER_DEFAULT)!!
 
-        return document.select("div.anime_muti_link > ul > li").parallelMap { server ->
-            runCatching {
-                val className = server.className()
-                if (!hosterSelection.contains(className)) return@runCatching emptyList()
-                val serverUrl = server.selectFirst("a")
-                    ?.attr("abs:data-video")
-                    ?: return@runCatching emptyList()
-
-                getHosterVideos(className, serverUrl)
-            }.getOrElse { emptyList() }
-        }.flatten().sort().ifEmpty { throw Exception("Failed to extract videos") }
+        return document.select("div.anime_muti_link > ul > li").parallelCatchingFlatMap { server ->
+            val className = server.className()
+            if (!hosterSelection.contains(className)) return@parallelCatchingFlatMap emptyList()
+            val serverUrl = server.selectFirst("a")
+                ?.attr("abs:data-video")
+                ?: return@parallelCatchingFlatMap emptyList()
+
+            getHosterVideos(className, serverUrl)
+        }.sort().ifEmpty { throw Exception("Failed to extract videos") }
```

Comment:
Thanks to your addition to the main app, the ifEmpty check is no longer needed

Reply:
Since you're overriding `videoListParse` and not `fetchVideoList`, adding `.sort()` isn't necessary either